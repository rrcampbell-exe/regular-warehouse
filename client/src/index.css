body {
  margin: 0;
  font-family: 'Roboto', sans-serif;;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.regular-video-visible {
  display: flex
}

.regular-video-invisible {
  display: none
}

.modal-container-invisible {
  display: none
}

.modal-container-visible {
  display: flex;
  position: fixed;
  top: 50%;
  left: 50%;
  right: auto;
  bottom: auto;
  transform: translate(-50%, -50%);
  border: solid black;
  padding: -10%, 0, -10%, 0;
  z-index: 9999;
  max-height: 80vh;
  background-color: white;
  border-radius: 0.25rem;
}

header {
  display: flex;
  justify-content: space-between;
}

header > a {
  margin: 20px;
}

.background {
  height: 100vh;
  background-image: url('https://images.pexels.com/photos/221047/pexels-photo-221047.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940');
  background-size: cover;
}

.op {
  opacity: 0.9;
  /* z-index: 1; */
}

.cat-container, .modal-text {
  display: flex;
  justify-content: center;
  align-items: center;
}

.secret-cat {
  max-width: 100%;
  max-height: 60%;
}

.modal-text {
  flex-direction: column;
}

.no-underline {
  text-decoration: none;
}

.text-shadow {
  text-shadow: 2px 2px 4px rgba(0,0,0, 0.25);
}