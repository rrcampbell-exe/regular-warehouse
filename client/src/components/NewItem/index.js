import { useMutation } from "@apollo/client";
import React, { useState } from "react";
import { ADD_ITEM } from "../../utils/mutations";
import { QUERY_ALL_ITEMS } from "../../utils/queries";

const faker = require("faker/locale/en_US");

let newPart_number; // Setting this variable in a larger scope, so mutliple functions can access it

function NewItem() {
  const [formData, setFormData] = useState({
    name: "",
    part_number: "",
    quantity: "",
  });
  const [addItem, { error }] = useMutation(ADD_ITEM, {
    update(cache, { data: { addItem } }) {
      try {
        const { items } = cache.readQuery({ query: QUERY_ALL_ITEMS });
        console.log(items[0].part_number);
        cache.writeQuery({
          query: QUERY_ALL_ITEMS,
          data: { items: [addItem, ...items] },
        });
      } catch (e) {
        console.error(e);
      }
    },
  });

  // const { data } = useQuery(QUERY_ALL_ITEMS); // To use in the generatetPartNumber function
  // console.log(data.items)
  function generatePartNumber() {
    if (!formData.part_number) {
      // create part number with faker
      const pnProto = faker.datatype.number({
        min: 100000,
        max: 199999,
      });

      let pnProtoArr = pnProto.toString().split("");
      pnProtoArr.splice(3, 0, "-");
      newPart_number = pnProtoArr.join("");

      // console.log(data.items)
      // for (let i = 0; i < data.items.length; i++) {  // Compare newPart_number with all others
      //     if (data.items[i].part_number === newPart_number) {  // If there's a match, rerun function to generate new number
      //         return generatePartNumber();
      //     }
      // }
    } else {
      // for (let i = 0; i < data.items.length; i++) {  // Compare newPart_number with all others
      //     if (data.items[i].part_number === newPart_number) {  // If there's a match, tell the user
      //         console.log("This part number is already in use")
      //         return; // Just return?
      //     }
      // }
      newPart_number = formData.part_number;
    }
  }

  const handleChange = (event) => {
    // Add an if statement to limit the length if necessary. If so, add characterCount state
    const { name, value } = event.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };
  const handleFormSubmit = async (event) => {
    event.preventDefault();
    // console.log(formData.part_number)

    generatePartNumber();

    try {
      // add item to database
      await addItem({
        variables: {
          name: formData.name,
          part_number: newPart_number,
          quantity: parseInt(formData.quantity),
        },
      });

      // Double check that this works (i'm suspicious)
      setFormData({ name: "", part_number: "", quantity: "" });
      // console.log(formData);
    } catch (e) {
      console.error(e);
    }
  };

  return (
    <div className='container-fluid shadow w-75 mt-2 p-3 rounded bg-light'>
      <h4>Add New Item</h4>
      <form onSubmit={handleFormSubmit} >
        <div className='mb-3'>
            <label htmlFor="name" className="form-label">Item Name:</label>
            <input
              placeholder="Item Name"
              name="name"
              type="text"
              value={formData.name} // -- is this necessary?
              onChange={handleChange}
              className="form-control"
            />
        </div>
       <div className='mb-3'>
          <label htmlFor="quantity" className="form-label">Item Quantity:</label>
            <input
              placeholder="0"
              name="quantity"
              type="number"
              value={formData.quantity}
              onChange={handleChange} // -- is this necessary?
              className="form-control"
            />
       </div>
       <div className='mb-3'>
            <label htmlFor="part_number" className="form-label">Part Number:</label>
              <input
                placeholder="000-000"
                name="part_number"
                type="text"
                value={formData.part_number}
                onChange={handleChange}
                className="form-control"
              />
        <p className="form-text">A new part number will be created if this space is left blank.</p>
       </div>
        <button className="btn btn-outline-secondary shadow-sm"><span className="material-icons md-12 align-bottom">add</span>Add item</button>
      </form>
      <p>
        {error && (
          <span>
            Something irregular has occurred... ðŸ¤” Are you sure you've entered a
            unique part number?
          </span>
        )}
      </p>
    </div>
  );
}

export default NewItem;
