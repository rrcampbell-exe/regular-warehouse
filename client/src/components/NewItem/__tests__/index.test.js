import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NewItem from '..';
import { ApolloProvider } from '@apollo/client';

afterEach(cleanup);

describe('NewItem component', () => {
    it('renders', () => { // this does not pass
        render(<NewItem/>)
    })
})