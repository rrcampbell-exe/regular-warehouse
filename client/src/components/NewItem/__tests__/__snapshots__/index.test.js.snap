// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewItem component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Add New Item
    </h2>
    <form>
      <label
        for="name"
      >
        Item Name:
      </label>
      <input
        name="name"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <label
        for="quantity"
      >
        Item Quantity:
      </label>
      <input
        name="quantity"
        placeholder="0"
        type="number"
        value=""
      />
      <label
        for="part_number"
      >
        Part Number:
      </label>
      <input
        name="part_number"
        placeholder="000-000"
        type="text"
        value=""
      />
      <p>
        A new part number will be created if this space is left blank.
      </p>
      <button>
        Add item
      </button>
    </form>
    <p />
  </div>
</DocumentFragment>
`;
